agglayer/grpc/agglayer_grpc_client.go

```diff
@@ -213,45 +218,48 @@ func convertProtoCertificateHeader(response *v1Types.CertificateHeader) *types.C
 }
 
 // convertToProtoBridgeExit converts a bridge exit to a proto bridge exit
-func convertToProtoBridgeExit(be *types.BridgeExit) *v1Types.BridgeExit {
+func convertToProtoBridgeExit(be *types.BridgeExit) *v1types.BridgeExit {
 	if be == nil {
 		return nil
 	}
 
-	protoBridgeExit := &v1Types.BridgeExit{
+	protoBridgeExit := &v1types.BridgeExit{
 		LeafType:    leafTypeToProto(be.LeafType),
 		DestNetwork: be.DestinationNetwork,
-		DestAddress: &v1Types.FixedBytes20{
+		DestAddress: &v1types.FixedBytes20{
 			Value: be.DestinationAddress.Bytes(),
 		},
-		Amount: &v1Types.FixedBytes32{
-			Value: common.BigToHash(be.Amount).Bytes(),
-		},
-		TokenInfo: &v1Types.TokenInfo{
+		TokenInfo: &v1types.TokenInfo{
 			OriginNetwork: be.TokenInfo.OriginNetwork,
-			OriginTokenAddress: &v1Types.FixedBytes20{
+			OriginTokenAddress: &v1types.FixedBytes20{
 				Value: be.TokenInfo.OriginTokenAddress.Bytes(),
 			},
 		},
 	}
 
+	if be.Amount != nil {
+		protoBridgeExit.Amount = &v1types.FixedBytes32{
+			Value: common.BigToHash(be.Amount).Bytes(),
```

Comment:
Optional nitpick, but for sake of semantical correctness (although I'm aware that under the hood, this is the same), maybe we can add the following function to aggkit's common package, and use it here, wdyt?:

```go
func BigIntToBytes32(n *big.Int) [32]byte {
	var b32 [32]byte
	bytes := n.FillBytes(make([]byte, 32)) // fills a 32-byte slice with big-endian representation
	copy(b32[:], bytes)
	return b32
}

```

Reply:
Its nice, but, why implement something we already have. As far as for logic is concerned, I checked it, the existing logic is not more complex than this one.